### PR TITLE
implement getContactsByPhoneNumber [DONE]

### PR DESCRIPTION
**Pull request by hatemalimam**
as discussed here https://github.com/rt2zz/react-native-contacts/issues/137#issuecomment-515280224
I'm submitting the PR that implements  `getContactsByPhoneNumber` on both platforms.

the changes are tested and can be tested on this branch https://github.com/hatemalimam/react-native-contacts-test/tree/feature/getContactsByPhoneNumber

### CODE REVIEW

**Comment by morenoh149** [comment]
Awesome work. The test repo has actually been deprecated (I'll archive it now). Instead demonstrate how to use this new api in the example https://github.com/rt2zz/react-native-contacts/blob/master/example/App.js

**Comment by hatemalimam** [comment]
I have updated the example app.
I hope you don't mind, I have taken the liberty to revamp the example app a bit, since it was very basic.

It includes the basic functionalities of this lib.

I have updated also the README accordingly.

The app is tested against iOS and android.

react-native-contacts

File: android/src/main/java/com/rt2zz/reactnativecontacts/ContactsManager.java [hunk 1]
```diff
@@ -138,6 +138,38 @@ protected Void doInBackground(final Void ... params) {
...
+
+    /**
+     * Retrieves contacts matching a phone number.
+     * Uses raw URI when <code>rawUri</code> is <code>true</code>, makes assets copy otherwise.
+     *
+     * @param phoneNumber phone number to match
+     * @param callback user provided callback to run at completion
+     */
+    private void getAllContactsByPhoneNumber(final String phoneNumber, final Callback callback) {
+        AsyncTask<Void,Void,Void> myAsyncTask = new AsyncTask<Void,Void,Void>() {
```
Comment (morenoh149) [inline android/src/main/java/com/rt2zz/reactnativecontacts/ContactsManager.java]: This function body should be inlined into `getContactsByPhoneNumber` on line 147. I think `getAllContactsMatchingString` was split into two function due to misunderstanding what was going on in `getAll`,`getAllWithoutPhotos` and `getAllContacts`.

@hatemalimam is it necessary for `getAllContactsByPhoneNumber` to be two functions? one decorated by `@ReactMethod` and one not decorated.

**Comment by morenoh149** [comment]
thanks!

**Comment by morenoh149** [comment]
I tried using the readme images but couldn't get them to work. If you want to resubmit them feel free.

File: android/src/main/java/com/rt2zz/reactnativecontacts/ContactsManager.java (hunk 1 above)
Comment (hatemalimam) [reply android/src/main/java/com/rt2zz/reactnativecontacts/ContactsManager.java]: In fact I see no reason to have them split like that, I was just following the one done in `getAllContactsMatchingString`, as it implies some convention.

I'll submit a refactor PR.

Comment (morenoh149) [reply android/src/main/java/com/rt2zz/reactnativecontacts/ContactsManager.java]: awesome! Things like this happen when multiple people work on a codebase, you don't remember why something was done the way it was. And the comments don't explain either.

**Comment by jamesawer3** [comment]
Is getContactsByPhoneNumber working in iOS ??? For me its not working updated react-native-contacts to latest version and tried but its saying getContactsByPhoneNumber not defined.

**Comment by hatemalimam** [comment]
I guess it's not released yet, not sure why, the current version is `5.0.2`, it's supposed to be released with `5.0.3`. however you can always add the dependency from the master branch if it's needed critically. 

@morenoh149 any plans on a soon release ?

**Comment by morenoh149** [comment]
I cut a release now.

**Comment by prozwang** [comment]
awesome ! cool ! Yet, there still one place need to be optimized
search for phone number is not good to use
if i want to find a friend in my contacts book, but i only remember part of his phone number 
i can't find him except i input his whole phone number

e.g.   Name: Tom.  PhoneNumber: 888812345
only input '888812345' ,i can find tom
but i expect when i input '888' or '12345' , my search list would show tom

**Comment by hatemalimam** [comment]
Well in android it would work as you've expected, but the provided api at the iOS side has an exact match, maybe I'll look into it later to see if we can support the same in iOS.

**Pull request merged**